crates/xlayer/db/src/internal_transaction_inspector.rs

```diff
@@ -0,0 +1,345 @@
+                self.sibling_count.truncate(depth + 1);
+            }
+            Ordering::Equal => {}
+        }
+        self.last_depth = depth;
+
+        let internal_index = self.sibling_count[depth];
+        let trace_index = self.traces.len() - 1;
+        self.trace_stack.push(trace_index);
+
+        let txn = &mut self.traces[trace_index];
+        txn.dept = depth as u64;
+        txn.internal_index = internal_index as u64;
+
+        self.sibling_count[depth] += 1;
+
+        self.current_path.push(internal_index);
+        if self.current_path.len() > 1 {
+            let suffix =
+                self.current_path[1..].iter().map(|s| s.to_string()).collect::<Vec<_>>().join("_");
+
+            txn.name.reserve(1 + suffix.len());
+            txn.name.push('_');
+            txn.name.push_str(&suffix);
+        }
+
+        txn.name = txn.call_type.clone() + &txn.name;
+    }
+
+    fn after_op(&mut self) {
+        self.current_path.pop();
+        if self.trace_stack.is_empty() {
+            self.all_traces.push(self.traces.clone());
+            self.reset();
+        }
+    }
+
+    pub fn get(&mut self) -> Vec<Vec<InternalTransaction>> {
+        self.all_traces.clone()
+    }
+
+    pub fn reset(&mut self) {
+        self.traces.clear();
+        self.current_path.clear();
+        self.last_depth = 0;
+        self.sibling_count = vec![0];
+        self.trace_stack.clear();
+    }
+}
+
+impl<CTX> Inspector<CTX, EthInterpreter> for TraceCollector
+where
+    CTX: ContextTr,
+{
+    fn call(&mut self, ctx: &mut CTX, inputs: &mut CallInputs) -> Option<CallOutcome> {
+        let call_type = match inputs.scheme {
+            reth_revm::interpreter::CallScheme::Call => "call",
+            reth_revm::interpreter::CallScheme::CallCode => "callcode",
+            reth_revm::interpreter::CallScheme::DelegateCall => "delegatecall",
+            reth_revm::interpreter::CallScheme::StaticCall => "staticcall",
+        }
+        .to_string();
+
+        let call_input = match &inputs.input {
+            CallInput::SharedBuffer(range) => ctx
+                .local()
+                .shared_memory_buffer_slice(range.clone())
+                .map(|s| Bytes::from(s.to_vec()))
+                .unwrap_or_default(),
+            CallInput::Bytes(b) => b.clone(),
+        };
+
+        self.init_op(
+            call_type,
+            inputs.caller.to_string(),
+            inputs.target_address.to_string(),
+            call_input,
+            inputs.value.get().to_string(),
+            inputs.gas_limit,
+            inputs.bytecode_address.to_string(),
+        );
+
+        self.before_op();
+
+        None
+    }
+
+    fn call_end(&mut self, _ctx: &mut CTX, _inputs: &CallInputs, outcome: &mut CallOutcome) {
+        let trace_index = self.trace_stack.pop().unwrap_or_default();
+        let (_before, after) = self.traces.split_at_mut(trace_index);
+        let (txn, remainder) = after.split_first_mut().unwrap();
```

Comment:
it's dangerous to use unwarp in rust, recommend to use this code style

let (txn, remainder) = after.split_first_mut()
    .ok_or_else(|| eyre::eyre!("Invalid trace stack state"))?;

Reply:
please fix all the use case of unwarp funcs